src/utils/logger.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <fmt/printf.h>
+
+extern int (*nrnpy_pr_stdoe_callback)(int, char*);
+
+template <typename... T>
+int Fprintf(FILE* stream, const char* fmt, T... args) {
+    if (nrnpy_pr_stdoe_callback && (stream == stdout || stream == stderr)) {
+        std::string message = fmt::sprintf(fmt, args...);
```

Comment:
Can you convert pointers to `fmt::ptr` here? Something like:
```
template<class T, class SFINAE = void>
T decorate_pointers(T t) {
  return t;
}

template<class T>
auto decorate_pointers<T, enable_if_t<is_pointer_v<T>>>(T* t) -> ... {
  return fmt::ptr(t);
}
```
Then replace `args...` with `decorate_pointers(args)...`.

Reply:
no, because for example fmt modify on the fly `char*` to print a string, this kind of thing. I understood this is the reason of this (boring) `fmt::ptr`.

Reply:
I see.